graylog2-server/src/main/java/org/graylog2/users/RoleImpl.java

```diff
@@ -0,0 +1,103 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog2.users;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.graylog2.shared.users.Role;
+
+import javax.annotation.Nullable;
+import javax.persistence.Id;
+import javax.validation.constraints.NotNull;
+import java.util.Set;
+
+public class RoleImpl implements Role {
```

Comment:
Why not use `@AutoValue` here? This would also take care of the (now missing) `Object#hashcode()` and `Object#equals()` implementations.